Adding evaluation to Mistral agent examples via Phoenix

Jgilhuly opened the pull request:
# Cookbook Pull Request

## Description

This PR adds tracing and evaluation walkthroughs to two of the Mistral agent examples. The changes are contained to optional sections at the end of the two notebooks, and give a comprehensive demo of how the Agents can be evaluating using Phoenix
...

## Type of Change

What type of PR is it?

- [ ] New Cookbook
  - [ ] Notebook File
    - [ ] Does it work on google colab?
  - [ ] Markdown File
- [x] Cookbook Update
  - [ ] Code Refactoring
  - [ ] Bug Fix
- [ ] README.md Update
___
- [ ] Other (please describe):

## Cookbook Checklist:

- [x] My code is easy to read and well structured.
- [x] I've tagged the versions of any dependency required.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generate no new warnings or errors.
___
- [ ] My changes do not concern the cookbooks.

## README.md Checklist

- [ ] I've added my cookbook to the table.
___
- [x] My changes do not concern the README file.

## Additional Context

*Add any other context or screenshots about the feature request here.*

...

sophiamyang commented:
Hi @Jgilhuly thanks so much for the PR! Would it be possible for you to add new agent notebooks to the Phoenix folder? Thanks!

Jgilhuly commented:
@sophiamyang moved the examples over to the Phoenix folder, let me know if you had something else in mind!

sophiamyang commented:
Thank you @Jgilhuly! Is it intended you have PHOENIX_API_KEY in the notebook? Is it okay for users to use this key or would you like to remove it and delete the key?

sophiamyang commented:
Thank you @Jgilhuly ! I got error message `ImportError: `llama-index-readers-file` package not found` running `arize_phoenix_evaluate_rag.ipynb`. could you help update please? Thank you!

Jgilhuly commented:
@sophiamyang - apologies for the delay. I fixed the bug you mentioned there, but there's an issue we still need to fix in the third_party/Phoenix/arize_phoenix_evaluate_rag.ipynb notebook to make sure our mistral eval model object is compatible with mistral>=1.0.0. I'll follow up as soon as we fix the issue, thanks for your patience here!

sophiamyang commented:
Thank you so much @Jgilhuly! Appreciate the help!

Jgilhuly commented:
@sophiamyang - apologies for the delay, we got the update made and I've fixed the notebooks here. I also went ahead and removed one since the other agent eval notebook covered the same topics, plus more.

sophiamyang commented:
Amazing thanks! Will take a look after the holidays and post on social.

Jgilhuly commented:
Hey @sophiamyang - checking back in here, let me know if there are any other updates I should make. Thanks for all your help!
